etc/ROS2TEST_QOS_PROFILES.xml

```diff
@@ -52,44 +53,55 @@ xsi:noNamespaceSchemaLocation="http://community.rti.com/schema/5.3.1/rti_dds_qos
             <element>
               <name>dds.transport.UDPv4.builtin.recv_socket_buffer_size</name>
               <value>65535</value>
-            </element -->
-            
+            </element>
+
             <!-- === SHMEM Transport ====================================================================== -->
-            <!-- maximum message size >
+            <!-- maximum message size -->
             <element>
-              <name>dds.transport.shmem.builtin.received_message_count_max </name>
-              <value>512</value>
+              <name>dds.transport.shmem.builtin.parent.message_size_max </name>
+              <value>1048576</value>
             </element>
             <element>
               <name>dds.transport.shmem.builtin.receive_buffer_size </name>
-              <value>1048576</value>
-            </element -->
-            
+              <value>16777216</value>
+            </element>
+            <element>
+              <name>dds.transport.shmem.builtin.received_message_count_max</name>
+              <value>16</value>
+            </element>
+
             <!-- allow/deny lists -->
             <!-- we try to exclude all vmware virtual network interfaces -->
-            <!-- DISABLED element>
+            <!-- <element>
               <name>dds.transport.UDPv4.builtin.parent.deny_interfaces</name>
               <value>FOOBAR,192.168.0.11,192.168.99.99</value>
             </element>
             <element>
               <name>dds.transport.UDPv4.builtin.parent.deny_multicast_interfaces</name>
               <value>FOOBAR,192.168.0.11,192.168.99.99</value>
-            </element -->
-            <!-- DISABLED element>
+            </element>
+            <element>
               <name>dds.transport.UDPv4.builtin.parent.allow_interfaces</name>
               <value>FOOBAR,192.168.112.2</value>
             </element>
             <element>
               <name>dds.transport.UDPv4.builtin.parent.allow_multicast_interfaces</name>
               <value>FOOBAR,192.168.112.2</value>
-            </element -->
+            </element> -->
           </value>
         </property>
       </participant_qos>
       <publisher_qos>
       </publisher_qos>
       <subscriber_qos>
       </subscriber_qos>
+
+      <datawriter_qos>
+        <publish_mode>
+          <kind>ASYNCHRONOUS_PUBLISH_MODE_QOS</kind>
+        </publish_mode>
+      </datawriter_qos>
```

Comment:
I added this, because I saw messages about asynchronous publishing not being enabled when uncommenting the `transport_builtin` section.

I think that was just a weird error message.
I've uncommented this from the final file without observing problems.

We're forcing this in the code option in code: https://github.com/ros2/rmw_connext/blob/c27c246b542d2980ed3fb039884cb34bc6dee8a6/rmw_connext_shared_cpp/src/qos.cpp#L221.